File: problems/1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
const maxVowels = (s, k) => {
    let max = 0;
    const vowelsSet = new Set(['a', 'e', 'i', 'o','u']);
    let count = 0;
    for (let i = 0; i < k; i++) {
        if(vowelsSet.has(s[i])) count ++;
    }
    max = count
    let l = 0;
    for (let r = l + k; r < s.length; r++) {
        if (vowelsSet.has(s[l])) {
            count --;
        }
        if (vowelsSet.has(s[r])) {
            count ++;
        }
        l ++;
        max = Math.max(max, count);
    }
    return max;
};

File: problems/1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        vowels = ['a', 'e', 'i', 'o','u']
        count = 0
        for cha in s[:k]:
            if cha in vowels:
                count += 1
        maxCount = count
        for i in range(len(s) - k):
            if s[i] in vowels:
                count -= 1
            if s[i + k] in vowels:
                count += 1
            maxCount = max(maxCount, count)
        return maxCount